### ISSUE TITLE
Inconsistent behavior of a VM with DV

### ISSUE THREAD

**Issue by vasiliy-ul** [kubevirt/containerized-data-importer#3028]
Not 100% sure if it is a CDI or KubeVIrt bug.

**What happened**:

One of the KubeVirt tests sometimes fails when I run it in a downstream test cluster:
https://github.com/kubevirt/kubevirt/blob/3cd65bd1f39e20200ce0bef46774eb6ee1db0bea/tests/storage/datavolume.go#L496
```go
It("[test_id:3190]should correctly handle eventually consistent DataVolumes", func() {
```

In short, the test does the following:
1. it creates a DV with an invalid registry URL
2. it runs a VM referencing that DV and ensures the VMI stays in `Pending` state (since the DV cannot be imported)
3. it makes the invalid registry reachable by creating a service of type `ExternalName` pointing to the correct registry
4. it ensures the DV/PVC gets populated and the VM starts normally

In the **upstream** KubeVirt CI, the test works fine. It uses a `RWO` `FileSystem` PVC which is always provisioned by `local` storage (according to the config):
https://github.com/kubevirt/kubevirt/blob/3cd65bd1f39e20200ce0bef46774eb6ee1db0bea/tests/default-ceph-config.json#L6C40-L6C40
```json
{
    ...
    "storageRWOFileSystem":    "local",
    ...
}
``` 
Since there is no CSI driver for `local` storage, the DV always gets the annotation 
```yaml
  annotations:
    cdi.kubevirt.io/storage.usePopulator: "false"
```

In the **downstream** test cluster, I use another storage provisioner (Longhorn). It has a CSI driver, hence the DV is annotated as:
```yaml
  annotations:
    cdi.kubevirt.io/storage.usePopulator: "true"
```

For some reason, that changes the behavior and the VMI transitions from `Pending` state to `Scheduling`. Thus, the test fails since it expects `Pending`:
https://github.com/kubevirt/kubevirt/blob/3cd65bd1f39e20200ce0bef46774eb6ee1db0bea/tests/storage/datavolume.go#L539
```go
Eventually(ThisVMIWith(vm.Namespace, vm.Name), 100).Should(BeInPhase(v1.Pending))
```

**What you expected to happen**:

I would expect consistent behavior irrespective of whether populators are used or not.

**How to reproduce it (as minimally and precisely as possible)**:

This can be reproduced with the KubeVirt local cluster by using the following DV and VM definitions:

<details>
  <summary>dv.yaml</summary>

  ```yaml
  apiVersion: cdi.kubevirt.io/v1beta1
  kind: DataVolume
  metadata:
    annotations:
      cdi.kubevirt.io/storage.usePopulator: "true"
    name: testdv
  spec:
    pvc:
      accessModes:
      - ReadWriteOnce
      resources:
        requests:
          storage: 512Mi
      storageClassName: rook-ceph-block-wffc
    source:
      registry:
        pullMethod: pod
        url: docker://fakeregistry:5000/kubevirt/cirros-container-disk-demo:devel
  ```
</details>

<details>
  <summary>vm.yaml</summary>

```yaml
apiVersion: kubevirt.io/v1
kind: VirtualMachine
metadata:
  name: testvmi
spec:
  running: true
  template:
    metadata:
      name: testvmi
    spec:
      architecture: amd64
      domain:
        devices:
          disks:
          - disk:
              bus: virtio
            name: disk0
          interfaces:
          - masquerade: {}
            name: default
        machine:
          type: q35
        resources:
          requests:
            memory: 128Mi
      networks:
      - name: default
        pod: {}
      terminationGracePeriodSeconds: 0
      volumes:
      - dataVolume:
          name: testdv
        name: disk0
```
</details>

```console
$ export KUBEVIRT_STORAGE=rook-ceph-default
$ make cluster-up
$ make cluster-sync
$
$ # with cdi.kubevirt.io/storage.usePopulator: "true"
$ k apply -f dv.yaml
$ k apply -f vm.yaml
$ k get vmi -w
NAME      AGE   PHASE     IP    NODENAME   READY
testvmi   0s    Pending                    
testvmi   1s    Pending                    
testvmi   1s    Pending                    False
testvmi   1s    Scheduling                    False
testvmi   1s    Scheduling                    False
$
$ # edit dv.yaml and set cdi.kubevirt.io/storage.usePopulator: "false"
$ k apply -f dv.yaml
$ k apply -f vm.yaml
$ k get vmi -w
NAME      AGE   PHASE     IP    NODENAME   READY
testvmi   0s    Pending                    
testvmi   0s    Pending                    
testvmi   0s    Pending                    False
testvmi   0s    Pending                    False
testvmi   1s    Pending                    False
testvmi   8s    Pending                    False
```

**Additional context**:

N/A

**Environment**:
- CDI version (use `kubectl get deployments cdi-deployment -o yaml`): `v1.58.0` (downstream cluster) or `v1.57.0` (KubeVirt cluster)
- Kubernetes version (use `kubectl version`): `v1.26.10` or `v1.27.6`
- DV specification: N/A
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): N/A
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Others: N/A

**Comment by akalenyu** [comment]
Thanks for reporting. We struggled to run this test downstream, see the first bullet point here:
https://github.com/kubevirt/kubevirt/pull/7005

I would say that if the eventual consistency is not broken, we should be okay with either of these states,
similar to what this PR does for populators:
https://github.com/kubevirt/kubevirt/pull/9918/commits/844e69b96a99e35955058bcd7fb5ef354d5ed701

**Comment by vasiliy-ul** [comment]
I remember I also spent some time in the past to properly setup this fakeregistry/registry matching.

I checked with the below diff and the test pass normally on my environment:

```diff
-                               Eventually(ThisVMIWith(vm.Namespace, vm.Name), 100).Should(BeInPhase(v1.Pending))
+                               Eventually(ThisVMIWith(vm.Namespace, vm.Name), 100).Should(Or(BeInPhase(v1.Pending), BeInPhase(v1.Scheduling)))
```

Basically that was my first idea to just submit that patch to KubeVirt tests. But then I thought it would be more appropriate to double-check here so we do not accidentally hide a potential bug. According to my observation, everything looks sane and there is no observable breakage. So if `Scheduling` state is also fine, then probably I can submit a fix for the test itself.
